### PR TITLE
Add parallel worker timings for remembered set scan

### PR DESCRIPTION
**Pull request by earthling-amzn**
### Description
Followed the existing pattern to add parallel working timings for remembered set scan.

### Testing
Ran Dacapo, looks like this in the logs:
```
[157.867s][202297][info] Concurrent Reset                  94013 us
[157.867s][202297][info] Pause Init Mark (G)              579650 us
[157.867s][202297][info] Pause Init Mark (N)              579456 us
[157.867s][202297][info]   Scan Remembered Set            483166 us, parallelism: 3.91x
[157.867s][202297][info]     RS: <total>                 1891580 us
[157.867s][202297][info]     RS: Scan Clusters           1891580 us, workers (us): 481972, 468615, 457912, 483081, 
[157.867s][202297][info]   Update Region States               78 us
[157.867s][202297][info] Concurrent Mark Roots             18932 us, parallelism: 1.98x

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to 9cc042c8e3c05a45deb8e28b63613f273d995c29


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/shenandoah pull/27/head:pull/27` \
`$ git checkout pull/27`

Update a local copy of the PR: \
`$ git checkout pull/27` \
`$ git pull https://git.openjdk.java.net/shenandoah pull/27/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 27`

View PR using the GUI difftool: \
`$ git pr show -t 27`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/shenandoah/pull/27.diff">https://git.openjdk.java.net/shenandoah/pull/27.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/shenandoah/shenandoahPhaseTimings.hpp [hunk 1]
```diff
@@ -99,8 +102,8 @@ class outputStream;
   f(conc_rendezvous_roots,                          "Rendezvous")                      \
   f(conc_evac,                                      "Concurrent Evacuation")           \
                                                                                        \
-  f(init_update_refs_gross,                         "Pause Init  Update Refs (G)")      \
-  f(init_update_refs,                               "Pause Init  Update Refs (N)")      \
+  f(init_update_refs_gross,                         "Pause Init Update Refs (G)")      \
+  f(init_update_refs,                               "Pause Init Update Refs (N)")      \
```
Comment (shipilev) [inline src/hotspot/share/gc/shenandoah/shenandoahPhaseTimings.hpp:106]: Revert this change, we should clean this up upstream.

File: src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp [hunk 2]
```diff
@@ -95,19 +95,20 @@ void ShenandoahScanRememberedTask::work(uint worker_id) {
   // This sets up a thread local reference to the worker_id which is necessary
   // the weak reference processor.
   ShenandoahParallelWorkerSession worker_session(worker_id);
+  ShenandoahWorkerTimingsTracker x(ShenandoahPhaseTimings::init_scan_rset, ShenandoahPhaseTimings::ScanClusters, worker_id);
 
   ShenandoahObjToScanQueue* q = _queue_set->queue(worker_id);
   ShenandoahObjToScanQueue* old = _old_queue_set == NULL ? NULL : _old_queue_set->queue(worker_id);
   ShenandoahMarkRefsClosure<YOUNG> cl(q, _rp, old);
-  RememberedScanner *rs = ShenandoahHeap::heap()->card_scan();
+  RememberedScanner *scanner = ShenandoahHeap::heap()->card_scan();
```
Comment (shipilev) [inline src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp:103]: While you are at it, can you put `*` to lean to the type? `RememberedScanner* scanner`.

**Review by shipilev — APPROVED** [review]
This looks good. Consider fixing minor nits.

File: src/hotspot/share/gc/shenandoah/shenandoahPhaseTimings.hpp (hunk 1 above)
Comment (earthling-amzn) [reply src/hotspot/share/gc/shenandoah/shenandoahPhaseTimings.hpp]: Created a pull request for this: https://github.com/openjdk/jdk/pull/3342
